includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -983,7 +983,12 @@ public function getAsJson() {
 			}
 
 			if ( $redirect !== 'no' && $article->getPage()->isRedirect() ) {
-				$article = Article::newFromTitle( $article->getPage()->followRedirect(), RequestContext::getMain() );
+				// this will be equal to false if redirect is not local
```

Comment:
Ugh, the [method's describtion](https://github.com/Wikia/app/blob/dev/includes/WikiPage.php#L661) says different :dango:

But when I read more carefully the [`getRedirectUrl()`](https://github.com/Wikia/app/blob/dev/includes/WikiPage.php#L674) method you hit line [676](https://github.com/Wikia/app/blob/dev/includes/WikiPage.php#L676) if it's local article and not a redirect or [707](https://github.com/Wikia/app/blob/dev/includes/WikiPage.php#L707) and then you can go with your first implementation in my opinion.

However, when I read the line [679](https://github.com/Wikia/app/blob/dev/includes/WikiPage.php#L679) and next one it doesn't make too much sense for me but seems like it's possible it'll return an URL. Therefore, I don't think it'll hurt us if we add the `!is_string()` since it's supposed to return a `Title` instance not title string.